src/test/ui/issues/issue-38091.stderr

```diff
@@ -1,8 +1,8 @@
 error[E0277]: the trait bound `(): Valid` is not satisfied
   --> $DIR/issue-38091.rs:8:5
    |
-LL | trait Iterate<'a> {
-   | ----------------- required by `Iterate`
+LL |     type Ty: Valid;
+   |     --------------- required by `Iterate::Ty`
```

Comment:
interesting that this changed

Reply:
This is because the def_id used in `ObligationCauseCode::ItemObligation` was changed